src/app/core/auth/services/auth.service.ts

```diff
@@ -15,4 +15,19 @@ export class AuthService {
       params: new HttpParams({ fromObject: credentials })
     })
   }
+
+  public register(registerData: RegisterData): Observable<RegisterData> {
+    const { name, email, password } = registerData
+    return this._http.post<RegisterData>('/api/auth/sign-up', {
+      name,
+      email,
+      password
+    })
+  }
+
+  public validateEmail(email: string): Observable<boolean> {
+    return this._http.post<boolean>(`/api/auth/sign-up/validate`, {
```

Comment:
This api endpoint does not return boolean. It either succeeds (200 OK) or fails (409 Conflict). It's **You**. who have to map the response too boolean depending on the outcome.